acorn/src/regexp.js

```diff
@@ -8,30 +8,40 @@ const pp = Parser.prototype
 export class RegExpValidationState {
   constructor(parser) {
     this.parser = parser
-    this.validFlags = `gim${parser.options.ecmaVersion >= 6 ? "uy" : ""}${parser.options.ecmaVersion >= 9 ? "s" : ""}${parser.options.ecmaVersion >= 13 ? "d" : ""}`
+    this.validFlags = `gim${parser.options.ecmaVersion >= 6 ? "uy" : ""}${parser.options.ecmaVersion >= 9 ? "s" : ""}${parser.options.ecmaVersion >= 13 ? "d" : ""}${parser.options.ecmaVersion >= 15 ? "v" : ""}`
     this.unicodeProperties = UNICODE_PROPERTY_VALUES[parser.options.ecmaVersion >= 14 ? 14 : parser.options.ecmaVersion]
     this.source = ""
     this.flags = ""
     this.start = 0
     this.switchU = false
+    this.switchV = false
     this.switchN = false
     this.pos = 0
     this.lastIntValue = 0
     this.lastStringValue = ""
     this.lastAssertionIsQuantifiable = false
+    this.lastMayContainStrings = false
```

Comment:
Would using return values on the set-expression-related methods, rather than a flag, be a workable way to track this? Looking at the code around this, tracking this value with a property seems verbose and difficult to see through.

Reply:
I will try to make that change. It will take some time, so please wait.

Reply:
I fixed that. https://github.com/acornjs/acorn/pull/1219/commits/d09dc4e187291677f6c4225c8606311026c162c1